src/eth_rpc/servers/kakarot_rpc.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    eth_provider::{provider::EthereumProvider, starknet::kakarot_core::to_starknet_transaction},
+    eth_rpc::api::kakarot_api::KakarotApiServer,
+};
+use jsonrpsee::core::{async_trait, RpcResult};
+use reth_primitives::B256;
+use starknet::{
+    core::{
+        crypto::compute_hash_on_elements,
+        types::{BroadcastedInvokeTransaction, BroadcastedInvokeTransactionV1, FieldElement},
+    },
+    providers::Provider,
+};
+use std::convert::TryInto;
+
+#[derive(Debug)]
+pub struct KakarotRpc<EP, SP> {
+    eth_provider: EP,
+    starknet_provider: SP,
+}
+
+impl<EP, SP> KakarotRpc<EP, SP> {
+    pub const fn new(eth_provider: EP, starknet_provider: SP) -> Self {
+        Self { eth_provider, starknet_provider }
+    }
+}
+trait ToElements {
+    fn try_into_v1(self) -> Result<BroadcastedInvokeTransactionV1, &'static str>;
+}
+
+impl ToElements for BroadcastedInvokeTransaction {
+    fn try_into_v1(self) -> Result<BroadcastedInvokeTransactionV1, &'static str> {
+        match self {
+            Self::V1(tx_v1) => Ok(tx_v1),
+            Self::V3(_) => Err("Transaction is V3, cannot convert to V1"),
+        }
+    }
+}
```

Comment:
Do you think this is possible to replace by a Rust `TryFrom` implementation like:

```rust
impl TryFrom<BroadcastedInvokeTransaction> for BroadcastedInvokeTransactionV1 {
    type Error = &'static str;

    fn try_from(tx: BroadcastedInvokeTransaction) -> Result<Self, Self::Error> {
        match tx {
            BroadcastedInvokeTransaction::V1(tx_v1) => Ok(tx_v1),
            BroadcastedInvokeTransaction::V3(_) => Err("Transaction is V3, cannot convert to V1"),
        }
    }
}
```

Reply:
also can you return `eyre::eyre!("Transaction is V3, cannot convert to V1")` for the error?

Reply:
Maybe I'm wrong, but `BroadcastedInvokeTransactionV1` is not defined in the current crate and only traits defined in the current crate can be implemented for types defined outside. So, I should define a new trait (like the `ToElements`) and insert this into it. But, if you think it makes more sense, I'll add it